pkg/build/api/validation/validation.go

```diff
@@ -103,11 +103,18 @@ func validateBuildSpec(spec *buildapi.BuildSpec, fldPath *field.Path) field.Erro
 	allErrs := field.ErrorList{}
 	s := spec.Strategy
 
-	if s.CustomStrategy == nil && spec.Source.Git == nil && spec.Source.Binary == nil && spec.Source.Dockerfile == nil {
+	if s.CustomStrategy == nil && s.JenkinsPipelineStrategy == nil && spec.Source.Git == nil && spec.Source.Binary == nil && spec.Source.Dockerfile == nil {
```

Comment:
this actually raises an interesting question.  If the Jenkinsfile is stored in the gitrepo, does it get reprocessed every time the build runs?  

When does the jenkins Job definition get updated?

what impact does that have if the job definition changes significantly between builds?